examples/pointcloud.js

```diff
@@ -0,0 +1,164 @@
+    edlGUI.add(edlRenderer.controlParameters, 'n').min(1).max(4).step(1)
+        .onChange(() => { view.notifyChange(true); });
+
+
+    view.render = () => {
+        const g = view.mainLoop.gfxEngine;
+        const r = g.renderer;
+
+        // update stats
+        info.textContent = `Nb points: ${pointcloud.counters.displayedCount.toLocaleString()} (${Math.floor(100 * pointcloud.counters.displayedCount / pointcloud.counters.pointCount)} %) (${view.mainLoop.gfxEngine.renderer.info.memory.geometries})`;
+
+        r.setClearColor(new itowns.THREE.Color(0xffffff));
+
+        if (!edlEnabled.value) {
+            // normal rendering
+            g.renderView(view);
+        } else {
+            edlRenderer.renderView(view);
+        }
+    };
+
+    const selectionGroup = new itowns.THREE.Group();
+
+    view.mainLoop.gfxEngine.renderer.domElement.addEventListener('dblclick', (event) => {
+        if (!pointcloud.root) {
+            return;
+        }
+        if (!selectionGroup.parent) {
+            // add selectionGroup to the scene after root, in order for selected points
+            // to be displayed after (alternative solution: use renderOrder)
+            view.scene.add(selectionGroup);
+            selectionGroup.frustumCulled = false;
+        }
+
+        selectionGroup.visible = false;
+        view.scene.traverse((o) => {
+            if (o.isPoints && o.baseId) {
+                o.material.enablePicking(true);
+            }
+        });
+
+        const mouse = {
+            x: event.offsetX,
+            y: (event.currentTarget.height || event.currentTarget.offsetHeight) - event.offsetY,
+        };
+
+        const buffer = view.mainLoop.gfxEngine.renderViewTobuffer(
+                view, view.mainLoop.gfxEngine.fullSizeRenderTarget,
+                mouse.x, mouse.y, 1, 1);
+
+        // see PointCloudProvider and the construction of unique_id
+        const objId = (buffer[0] << 8) | buffer[1];
+        const index = (buffer[2] << 8) | buffer[3];
+
+        const i = (objId << 16) | index;
+
+        view.scene.traverse((o) => {
+            if (o.isPoints && o.baseId) {
+                if (o.baseId == objId) {
+                    const positions = o.geometry.attributes.position.array;
+                    const p = positions.slice(3 * index + 0, 3 * index + 3);
+                    console.log('NEW SELECTION', p, o.owner.name);
+                    selectionGroup.layers.set(pointcloud.threejsLayer);
+                }
+
+                o.material.enablePicking(false);
+            }
+        });
+
+        selectionGroup.visible = true;
+        view.notifyChange(true);
+    });
+
+    const flyControls = new itowns.FlyControls(view, { focusOnClick: true });
+
+    menuGlobe.gui.add(pointcloud, 'octreeDepthLimit', -1, 20).name('Depth limit')
+        .onChange(() => view.notifyChange(true));
+    menuGlobe.gui.add(pointcloud, 'pointBudget', 500000, 15000000).name('Max point count')
```

Comment:
For all the pointclouds I tested, 500k was a very low value.
I replaced `500000` by `1` anyway.